csharp/src/Ice/SlicOptions.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) ZeroC, Inc. All rights reserved.
+
+using System;
+
+namespace ZeroC.Ice
+{
+    internal class SlicOptions
+    {
+        internal TimeSpan IdleTimeout { get; set; } = TimeSpan.FromSeconds(30);
+
+        internal int MaxBidirectionalStreams { get; set; } = 100;
+
+        internal int MaxUnidirectionalStreams { get; set; } = 100;
+
+        internal int PacketSize { get; set; } = 32 * 1024;
+    }
```

Comment:
Seems this can be a `record type` with init only properties

```
internal record SlicOptions
    {
        internal TimeSpan IdleTimeout { get; init; } = TimeSpan.FromSeconds(30);
        internal int MaxBidirectionalStreams { get; init; } = 100;
        internal int MaxUnidirectionalStreams { get; init; } = 100;
        internal int PacketSize { get; init; } = 32 * 1024;
    }
```

and use as

```
SlicOptions = new SlicOptions
{
    MaxBidirectionalStreams = GetPropertyAsInt("Ice.Slic.MaxBidirectionalStreams") ?? 100,
    MaxUnidirectionalStreams = GetPropertyAsInt("Ice.Slic.MaxBidirectionalStreams") ?? 100,
    IdleTimeout = GetPropertyAsTimeSpan("Ice.Slic.IdleTimeout") ?? TimeSpan.FromSeconds(30),
    PacketSize = GetPropertyAsInt("Ice.Slic.PacketSize") ?? 32 * 1024
};


bool serialized = adapter?.SerializeDispatch ?? false;
Options = new SlicOptions
{
    MaxBidirectionalStreams = serialized ? 1 : Endpoint.Communicator.SlicOptions.MaxBidirectionalStreams,
    MaxUnidirectionalStreams = serialized ? 1 : Endpoint.Communicator.SlicOptions.MaxUnidirectionalStreams,
    IdleTimeout = Endpoint.Communicator.SlicOptions.IdleTimeout,
};
```